### PR TITLE
Set required python version to the minimum version we support

### PR DESCRIPTION
**Pull request by kohsuke**
... to help catch problems earlier.

### CODE REVIEW

**Comment by kohsuke** [comment]
Given how this PR is going, I'm starting to get the sense that we have never actually tested with different Python versions before

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install pipenv
-        pipenv install --dev
+        pipenv install --dev --python ${{ matrix.python-version }}
```
Comment (Konboi) [inline .github/workflows/python-package.yml:31]: TIL: https://pipenv.pypa.io/en/latest/advanced/#pipenv.environments.PIPENV_DEFAULT_PYTHON_VERSION

**Review by Konboi — APPROVED** [review]
LGTM

Nice work!!

**Comment by ninjinkun** [comment]
Thanks!

**Pull request merged**